src/test/java/fi/tkgwf/ruuvi/common/ParserTest.java

```diff
@@ -0,0 +1,219 @@
+        data[0] = (byte) 0xAA;
+        data[1] = (byte) 0xFE;
+        data[2] = (byte) 0x10; // Frame type: URL
+        data[3] = (byte) 0xF8; // Power
+        data[4] = (byte) 0x03; // https://
+        data[5] = 'r';
+        data[6] = 'u';
+        data[7] = 'u';
+        data[8] = '.';
+        data[9] = 'v';
+        data[10] = 'i';
+        data[11] = '/';
+        data[12] = '#';
+        System.arraycopy(hashBytes, 0, data, 13, hashBytes.length);
+        return data;
+    }
+
+    private byte[] dataWithCompany(String rawHex) {
+        try {
+            byte[] rawBytes = Hex.decodeHex(rawHex);
+            byte[] data = new byte[2 + rawBytes.length];
+            data[0] = (byte) 0x99; // company id
+            data[1] = (byte) 0x04; // company id
+            System.arraycopy(rawBytes, 0, data, 2, rawBytes.length);
+            return data;
+        } catch (DecoderException e) {
+            Assert.fail();
+            return null;
+        }
+    }
+
+    @Before
+    public void setUp() {
+        this.parser = new AnyDataFormatParser();
+    }
+
+
+    /**
+     * test_decode_is_valid in https://github.com/ttu/ruuvitag-sensor/blob/master/tests/test_decoder.py
+     */
+    @Test
+    public void testDecodeEddystone() {
+        // https://ruu.vi/#AjwYAMFc
+        RuuviMeasurement m = parser.parse(eddystoneData("AjwYAMFc"));
+        assertEquals(24.0, m.getTemperature());
+        assertEquals(99500.0, m.getPressure());
+        assertEquals(30.0, m.getHumidity());
+        assertEquals((Integer) 2, m.getDataFormat());
+        assertNull(m.getAccelerationY());
+        assertNull(m.getAccelerationZ());
+        assertNull(m.getBatteryVoltage());
+        assertNull(m.getMeasurementSequenceNumber());
+        assertNull(m.getMovementCounter());
+        assertNull(m.getTxPower());
+    }
+
+
+    /**
+     * test_decode_is_valid_case in https://github.com/ttu/ruuvitag-sensor/blob/master/tests/test_decoder.py
+     */
+    @Test
+    public void testDecodeEddystone2() {
+        // https://ruu.vi/#AjgbAMFc
+        RuuviMeasurement m = parser.parse(eddystoneData("AjgbAMFc"));
+        assertEquals(27.0, m.getTemperature());
+        assertEquals(99500.0, m.getPressure());
+        assertEquals(28.0, m.getHumidity());
+        assertEquals((Integer) 2, m.getDataFormat());
+        assertNull(m.getAccelerationY());
+        assertNull(m.getAccelerationZ());
+        assertNull(m.getBatteryVoltage());
+        assertNull(m.getMeasurementSequenceNumber());
+        assertNull(m.getMovementCounter());
+        assertNull(m.getTxPower());
+    }
+
+    /**
+     * test_decode_is_valid_weatherstation_2017_04_12 in https://github.com/ttu/ruuvitag-sensor/blob/master/tests/test_decoder.py
+     */
+    @Test
+    public void testDecodeEddystone3() {
+        // https://ruu.vi/#AjUX1MAw0
+        RuuviMeasurement m = parser.parse(eddystoneData("AjUX1MAw0"));
+//        assertEquals(23.828125, m.getTemperature());
```

Comment:
I could not make this test pass -- any idea why it would be failing?

Reply:
..and the ruu.vi site ( https://ruu.vi/#AjUX1MAw0 ) parses that as 24 degrees, which is neither of those, even with any reasonable rounding.. 🤔 rather interesting, I'll take a closer look at this at some point